votor/src/certificate_pool.rs

```diff
@@ -1985,7 +1985,11 @@ mod tests {
             )
             .is_ok());
         let cert_4 = CertificateMessage {
-            certificate: Certificate::new(CertificateType::Finalize, 4, None),
```

Comment:
This is to test the case where we have Finalize but not accompanied Notarize certificate, I believe in that case you should not clean up this aggressively. (White paper page 43, "or a finalization certificate for s *together* with a notarization certificate for b in the same slots")

Reply:
reverted.